### ISSUE TITLE
Blas handle not initialized, API Error

### ISSUE THREAD

**Issue by Sentient07** [Theano/libgpuarray#375]
Hi, 
When I was training, I had an error in my error due to a shape mismatch. When I rectified the error and resumed the training, I faced this error before the device gets mapped. 
```
ERROR (theano.gpuarray): Could not initialize pygpu, support disabled
Traceback (most recent call last):
  File "/Users/Ramana/projects/theano/Theano/theano/gpuarray/__init__.py", line 164, in <module>
    use(config.device)
  File "/Users/Ramana/projects/theano/Theano/theano/gpuarray/__init__.py", line 151, in use
    init_dev(device)
  File "/Users/Ramana/projects/theano/Theano/theano/gpuarray/__init__.py", line 100, in init_dev
    pygpu.blas.gemm(0, tmp, tmp, 0, tmp, overwrite_c=True)
  File "pygpu/blas.pyx", line 129, in pygpu.blas.gemm (pygpu/blas.c:3354)
  File "pygpu/blas.pyx", line 44, in pygpu.blas.pygpu_blas_rgemm (pygpu/blas.c:2011)
GpuArrayException: ('Blas handle not initialized, API error.', 11)
Mapped name None to device cuda: GeForce GT 750M (0000:01:00.0)
E.
```
Continuation(Edit) [Sorry, I had finished the comment instead of moving to next line]
When I try to restart the training before waiting (~5minutes), I face this error. Before, when I used to run tests on the new backend, I reran the tests with few seconds gap and it was perfectly fine.

**Comment by nouiz** [comment]
Are you sure that no process stay in the back-ground that use the GPU? Can you try to reboot? Waiting should not fix anything and the time should not help Theano. So I'm puzled by this.

Note, you aren't the only one having this type of error, but we aren't able to reproduce it. Do you have multiple cuda version? If so, this could be part of the problem, make sure to have only 1 in your environment. We have suspison that the new back-end have problems when multiple cuda version are available in the environment.

**Comment by Sentient07** [comment]
> Are you sure that no process stay in the back-ground that use the GPU? Can you try to reboot? Waiting should not fix anything and the time should not help Theano. So I'm puzled by this.

This was the only process on the GPU after I stopped the script. I checked once before the subsequent run and it was the Intel GPU that was being used. I could not reproduce this when I tried it on the server that has K20 GPU(but CUDA 7.5). I don't know what to interpret from this, but,  if I preallocate memory (on my PC that has 750M, on which I had the error when I reported), it works fine without any error, and I can resume with few seconds of interval.  

> Note, you aren't the only one having this type of error, but we aren't able to reproduce it. Do you have multiple cuda version? 

I still haven't gotten rid of 7.5. ( I haven't deleted it). But that does not exist in the `$PATH` and `nvcc -V` points to `release 8.0, V8.0.46`.
